src/postprocessing/pp-av1.c

```diff
@@ -55,6 +55,10 @@ int janus_pp_av1_create(char *destination, char *metadata, gboolean faststart, c
 		return -1;
 	}
 
+	char filename[1024];
+	snprintf(filename, sizeof(filename), "%s", destination);
+	fctx->url = g_strdup(filename);
```

Comment:
I don't understand why you're going through a `sprintf` here. Can't you just `g_strdup(destination)`?

Reply:
You're right, thanks! Fixed.

Reply:
Don't forget the `g_free(fctx->url);` that the commit you referenced includes, or you'll be introducing a leak. I see that patch also did a few other things for each codec, like setting `CODEC_FLAG_GLOBAL_HEADER`: are those other changes unneeded for AV1 processing?

Reply:
If I am not wrong, the `avformat_free_context` function is used in `pp_av1.c` when closing the file, which already frees the memory allocated (according to [ffmpeg documentation](https://ffmpeg.org/doxygen/trunk/group__lavf__core.html#gac2990b13b68e831a408fce8e1d0d6445)), so there is no need for `g_free(fctx->url)`.

I think the patch just removed unused comments and that `CODEC_FLAG_GLOBAL_HEADER` flag was already there for `H264` and `H265` before the patch I mentionned. I can try to add the global header flag but I am not familiar enough with `libav` to know what will be the impact on the resulting video. I think it's not used either for WebM postprocessing.

Reply:
When I add `g_free` before `avformat_free_context`, I do get a `free(): double free detected in tcache 2` but it seems to be working fine if I add it after. I updated the pull request. 

What I could do is use `av_strdup` instead of `g_strdup` I guess.

Reply:
That's because you ignored part of my previous message :slightly_smiling_face: 

> You have to free it yourself with g_free and then set the pointer to NULL, which is what the other patch did.

You didn't set the pointer to `NULL`, so `avformat_free_context` tries to free something that doesn't exist anymore.

Reply:
Oh sorry, Monday mornings ... 😅 
I can move it up if you prefer.

Concerning the global flag headers, I suggest doing it in another pull request as it requires a bit more work (creating a codec context and switching to a codec parameter which is not yet done for in `pp-av1`)

Reply:
Sure, no problem! I asked about the global flag header only because it was in the other patch, and so I wondered if it could be needed for AV1 too: it may very well be that it isn't, I rarely work with any AV1 recording at all so I don't have experience with them.

Reply:
I'm new to AV1 recording, so I don't know if the global flag header helps but I'll take a deeper look at this as soon as I have time.